### PR TITLE
Cherry pick update to infinispan 11.0.11 back to jetty-9.4.x(#6766)

### PR DESCRIPTION
**Pull request by janbartel**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -995,6 +995,39 @@
...
+      <dependency>
+        <groupId>org.jboss.logmanager</groupId>
+        <artifactId>jboss-logmanager</artifactId>
+        <version>2.1.15.Final</version>
+      </dependency>
+      <dependency>
+        <groupId>org.jboss.threads</groupId>
+        <artifactId>jboss-threads</artifactId>
+        <version>3.1.0.Final</version>
+      </dependency>
```
Comment (joakime) [inline pom.xml:1035]: The dependencies in 9.4.x are sorted alphabetically.
(Soon the 10.0.x deps will be too)

File: jetty-infinispan/infinispan-common/pom.xml [hunk 2]
```diff
@@ -37,8 +37,13 @@
     <dependency>
       <groupId>org.infinispan</groupId>
       <artifactId>infinispan-core</artifactId>
-      <version>${infinispan.version}</version>
       <optional>true</optional>
+      <exclusions>
+          <exclusion>
```
Comment (olamy) [inline jetty-infinispan/infinispan-common/pom.xml]: nit but we usually have 2 spaces indent in pom

File: tests/test-sessions/test-infinispan-sessions/pom.xml [hunk 3]
```diff
@@ -108,6 +115,12 @@
       <artifactId>infinispan-core</artifactId>
       <version>${infinispan.version}</version>
       <scope>test</scope>
+      <exclusions>
+          <exclusion>
```
Comment (olamy) [inline tests/test-sessions/test-infinispan-sessions/pom.xml]: nit but we usually have 2 spaces indent in pom

File: tests/test-sessions/test-infinispan-sessions/src/test/java/org/eclipse/jetty/server/session/remote/RemoteInfinispanTestSupport.java [hunk 4]
```diff
@@ -67,18 +66,18 @@
...
                     ":" + infinispanVersion)
-                    .withEnv("APP_USER", "theuser")
-                    .withEnv("APP_PASS", "foobar")
-                    .withEnv("MGMT_USER", "admin")
-                    .withEnv("MGMT_PASS", "admin")
-                    .waitingFor(new LogMessageWaitStrategy()
-                        .withRegEx(".*Infinispan Server.*started in.*\\s"))
-                    .withExposedPorts(4712, 4713, 8088, 8089, 8443, 9990, 9993, 11211, 11222, 11223, 11224)
-                    .withLogConsumer(new Slf4jLogConsumer(INFINISPAN_LOG));
+                .withEnv("USER", "theuser")
```
Comment (olamy) [inline tests/test-sessions/test-infinispan-sessions/src/test/java/org/eclipse/jetty/server/session/remote/RemoteInfinispanTestSupport.java:73]: doh I can't believe checkstyle doesn't complain about removed/missing indent here.

**Review by olamy — DISMISSED** [review]
lgtm only nit picking comments on formatting

**Review by olamy — APPROVED** [review]

**Pull request merged**